substrate/frame/revive/src/exec.rs

```diff
@@ -586,6 +592,8 @@ struct Frame<T: Config> {
 	read_only: bool,
 	/// The caller of the currently executing frame which was spawned by `delegate_call`.
 	delegate_caller: Option<Origin<T>>,
+	/// The address of the contract the call was delegated to.
+	delegate_address: Option<H160>,
```

Comment:
Should be merged into one field.

Reply:
I am using the field to read immutable. It works as expected. https://github.com/paritytech/revive/pull/80/files#diff-30c3c30418972244514e3676b94ce9e7f41793080aad74989ad680f10d41ed26R46

Reply:
I am also trying to keep the changes at minimum

Reply:
I am looking at refactoring this so that frame can have `sender, caller and callee` but the problem is the root.
We're treating Root as mapped but unless we map it to some fixed address
```rust
fn ensure_mapped(&self) -> DispatchResult {
		match self {
			Self::Root => Ok(()),
```

Reply:
I think I figured it out. Are you open for a refactoring?

Reply:
It is fine to treat it is mapped. Because the contract will trap anyways when it tries to read the field.

Reply:
> I am also trying to keep the changes at minimum

If you add a new field you can as well add it to `delegate_caller` by adding a new structure. To signal that they are both either `None` or `Some`.